base/mpfr.jl

```diff
@@ -119,6 +119,9 @@ end
 convert(::Type{Rational}, x::BigFloat) = convert(Rational{BigInt}, x)
 convert(::Type{AbstractFloat}, x::BigInt) = BigFloat(x)
 
+float(::Type{BigInt})   = BigFloat
+float(::Type{BigFloat}) = BigFloat
```

Comment:
Not sure if such a method already exists, but something like this would be more general and would also cover this case:
```julia
float{T<:AbstractFloat}(::Type{T}) = T
```
If that method already exists then you should remove this one.